packages/react-switch/Spec.md

```diff
@@ -2,62 +2,135 @@
+| onAriaLabel  | @deprecated Use `ariaLabel` for name, and let the metadata convey state                         | deprecated                                         |
+| offAriaLabel | @deprecated Use `ariaLabel` for name, and let the metadata convey state                         | deprecated                                         |
+| inlineLabel  | Whether the label (not the onText/offText) should be positioned inline with the toggle control. | Toggle's label will be handled as a child element. |
+| onChanged    | @deprecated Use `onChange` instead.                                                             | deprecated                                         |
+| theme        | Theme provided by HOC.                                                                          | Not used in converged components                   |
+| styles       | Optional styles for the component.                                                              | Not used in converged components                   |
+| role         | Whether to use the 'switch' role (ARIA 1.1) or the 'checkbox' role (ARIA 1.0).                  | Toggle has a hidden input element.                 |
 
 ## Structure
 
 - _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
 
-## Migration
+```jsx=
+<Switch checked={true}/>
+```
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+- _**Internal**_
 
-- _Migration from v8_
-- _Migration from v0_
+```jsx=
+<slots.root {...slotProps.root}>
+  <slots.rail {...slotProps.rail} />
+  <slots.track {...slotProps.track} />
+  <slots.thumbWrapper {...slotProps.thumbWrapper}>
+    <slots.thumb {...slotProps.thumb} />
+  </slots.thumbWrapper>
+  <input type="checkbox" />
+</slots.root>;
+```
+
+- \_**DOM**
+
+```jsx=
+<div className="ms-switch-root">
+  <div className="ms-switch-rail" />
+  <div className="ms-switch-track" />
+  <div className="ms-switch-thumbWrapper">
+    <div className="ms-switch-thumb" />
+  </div>
+  <input type="checkbox" />
+</div>;
+```
 
 ## Behaviors
 
-_Explain how the component will behave in use, including:_
+### Component States
+
+- **Disabled**
+  - When disabled, all events are ignored, and the Switch's value never updates.
+  - Does not allow focus.
+- **Checked**
+  - Toggle is off when the “thumb” is indicated on the left.
+  - Toggle is on when the thumb is indicated on the right.
+  - This is switched in RTL.
+  - When switching toggles on and off, the on state should change using the checked state appearance styles.
+
+### Hover
+
+The cursor changes to the hand icon. The outline of the toggle and thumb should also darken. This helps reinforce that the area is interactive.
+
+### Keyboard
+
+Since the toggle is an interactive component, it must be focusable and keyboard accessible.
+The expected keyboard shortcut for activating a toggle is the Spacebar.
```

Comment:
nit:
```suggestion
Since the switch is an interactive component, it must be focusable and keyboard accessible.
The expected keyboard shortcut for activating a switch is the Spacebar key.
```